pkg/aws/metadata/metadata.go

```
@@ -11,53 +11,78 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 package metadata
 
 import (
+	"context"
 	"fmt"
 	"io"
-	"net/http"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
 )
 
-func getMetadata(name string) (string, error) {
-	resp, err := http.Get("http://169.254.169.254/latest/meta-data/" + name)
+func newClient() (*imds.Client, error) {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Printf("error: %v", err)
```

Let's just return the error without logging because the caller can see the error and log if they want to. This code will be retried anyway if it fails, so we don't want duplicate log msgs

---

done